osu.Game/Rulesets/Scoring/HitEvent.cs

```diff
@@ -50,15 +53,15 @@ public readonly struct HitEvent
         /// Creates a new <see cref="HitEvent"/>.
         /// </summary>
         /// <param name="timeOffset">The time offset from the end of <paramref name="hitObject"/> at which the event occurs.</param>
+        /// <param name="scaledTimeOffset">The rate-scaled <paramref name="timeOffset"/></param>
         /// <param name="result">The <see cref="HitResult"/>.</param>
-        /// <param name="gameplayRate">The true gameplay rate at the time of the event.</param>
         /// <param name="hitObject">The <see cref="HitObject"/> that triggered the event.</param>
         /// <param name="lastHitObject">The previous <see cref="HitObject"/>.</param>
         /// <param name="position">A position corresponding to the event.</param>
-        public HitEvent(double timeOffset, double? gameplayRate, HitResult result, HitObject hitObject, [CanBeNull] HitObject lastHitObject, [CanBeNull] Vector2? position)
+        public HitEvent(double timeOffset, double scaledTimeOffset, HitResult result, HitObject hitObject, [CanBeNull] HitObject lastHitObject, [CanBeNull] Vector2? position)
```

Comment:
This feels really bad. Looking at the diff at face value, is it not possible to just keep the `gameplayRate` variable but override `ManiaScoreProcessor` to always provide 1 instead of the actual gameplay rate itself? This of course comes with repurposing of the variable's name and docs so the code doesn't look weird. Something like renaming `gameplayRate` to `timeScale` or otherwise with explanatory docs.

Reply:
Your suggestion was my first approach but it felt weird due to it being called `gameplayRate`. Something like `timeScale` sounds a lot better.